portfolio/src/main/java/com/google/sps/servlets/NewMapMarkerServlet.java

```diff
@@ -45,22 +47,28 @@
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     try {
-      MapMarker newMapMarker = new MapMarker(
-        request.getParameter("collegeName"), 
-        request.getParameter("internName"),
-        (long) getLngLat(request.getParameter("collegeAddress")).geometry.location.lng,
-        (long) getLngLat(request.getParameter("collegeAddress")).geometry.location.lat);
+      MapMarker newMapMarker = new MapMarker(request.getParameter("collegeName"), request.getParameter("internName"),
+          (long) getLngLat(request.getParameter("collegeAddress")).geometry.location.lng,
+          (long) getLngLat(request.getParameter("collegeAddress")).geometry.location.lat);
       DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
       datastore.put(newMapMarker.makeEntity());
 
       response.sendRedirect("/index.html");
-    } catch (IOException | NotFoundException e) {
-        throw new IOException(e.getMessage());
-    }
+    } catch (ApiException e) {
+        response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+    } catch (IOException e) {
+        response.sendError(HttpServletResponse.SC_NOT_FOUND);
```

Comment:
Based on the documentation, it seems like an IOException does not indicate that the address could not be found: https://googlemaps.github.io/google-maps-services-java/v0.2.4/javadoc/com/google/maps/GeocodingApiRequest.html

This should probably be SC_INTERNAL_SERVER_ERROR like the others